table/block_based_table_reader.cc

```diff
@@ -813,7 +813,10 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
       CachableEntry<IndexReader> index_entry;
       unique_ptr<InternalIterator> iter(
           new_table->NewIndexIterator(ReadOptions(), nullptr, &index_entry));
-      index_entry.value->CacheDependencies(pin);
+      assert(index_entry.value != nullptr || !iter->status().ok());
+      if (index_entry.value != nullptr) {
+        index_entry.value->CacheDependencies(pin);
```

Comment:
Should we add the same check to filter_entry, index_reader, and filter below?

Reply:
It looks like those ones are safe from nullptr, either because they check `s.ok()` or because they explicitly check against nullptr.

Reply:
oh you are right. thanks.